ajo/in-app-messages/README.md

```diff
@@ -0,0 +1,77 @@
+# In-App Messages
+Web In-App Messaging is a feature that empowers marketers with the ability to deliver personalized messages to web users. It allows them to display personalized messages on websites using modal-overlay type messages. With this new feature, marketers can effectively engage with web customers, driving user engagement, retention, and conversions.
+
+## Overview
+
+This sample is using [Adobe Experience Platform Web SDK](https://experienceleague.adobe.com/docs/experience-platform/edge/home.html) to display in-app messages.
+
+This is how an in-app message looks like:
+<img src="../../.assets/in-app-message1.png" width="800"/>
+
+## Getting Started
+
+1. Author in-app message content in Adobe Journey Optimizer.
+2. Configure the sample to use your Adobe Journey Optimizer orgid,  datastreamId and surfaceUrl. Sample configuration is in the .env file.
+3. Run the sample.
+
+## Running the sample
+Prerequisite: [install node and npm](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm).
+
+To run this sample:
+1. [Setup local SSL certificates for https](https://github.com/adobe/alloy-samples/blob/main/LocalSSLCertificateSetup.md)
+2. Clone the repository to your local machine.
+3. Open a terminal and change directory to this sample's folder.
+4. Run `npm install`
+5. Run `npm start`
+6. Open a web browser to https://localhost
+
+## How it works
+
+There are two ways to display in-app message content from Adobe Journey Optimizer.
+
+[Alloy](https://experienceleague.adobe.com/docs/experience-platform/edge/home.html) is included on the page.
+1. On page load the sendEvent command is used to fetch in-app message content.
+   example :
+
+`alloy("sendEvent", {
+  renderDecisions: true,
+  personalization {
+   decisionContext: {
+      "user": "alloy-sample"
```

Comment:
I suggest removing `decisionContext` in the page load (sendEvent) use case and rely only on surface.

Add a `surfaces` array to the `personalization` object instead, and use a hashed surface name like `#welcome`.

```js
alloy("sendEvent", {
  renderDecisions: true,
  personalization: {
   surfaces: ['#welcome']
  }
});
```

In AJO, you will use `https://localhost#welcome` for the surface/URL to match ☝️

Reply:
done